src/runtime_src/core/tools/common/tests/TestVerify.cpp

```diff
@@ -16,7 +17,15 @@ TestVerify::TestVerify()
 boost::property_tree::ptree
 TestVerify::run(std::shared_ptr<xrt_core::device> dev)
 {
-  boost::property_tree::ptree ptree = get_test_header();
-  runTestCase(dev, "22_verify.py", ptree);
+  auto device_name = xrt_core::device_query_default<xrt_core::query::rom_vbnv>(dev, "");
+  boost::property_tree::ptree ptree;
+  if(device_name.find("IPU") != std::string::npos) {
+    //run ipu verify
+    ptree = std::make_shared<TestIPU>()->run(dev);
```

Comment:
Curious... I have the feeling that once the full expression is evaluated the shared pointer is removed.
If this is the correct expected behavior why writing such a code using dynamic memory and not just
```suggestion
    ptree = TestIPU {}.run(dev);
```
?

Reply:
Thanks for the suggestion. The shared pointer goes out of scope once the test is run. I've updated the code to what you suggested